components/script/dom/xrsession.rs

```diff
@@ -170,6 +171,50 @@ impl XRSession {
                 let event = XRSessionEvent::new(&self.global(), atom!("end"), false, false, self);
                 event.upcast::<Event>().fire(self.upcast());
             },
+            XREvent::Select(input, kind, frame) => {
+                let source = self
+                    .input_sources
+                    .borrow_mut()
+                    .iter()
+                    .find(|s| s.id() == input)
+                    .map(|x| DomRoot::from_ref(&**x));
+                if let Some(source) = source {
+                    let frame = XRFrame::new(&self.global(), self, frame);
```

Comment:
Do you have a spec link for this? Generating a new frame here seems odd, since we're not in an rAF.

Reply:
https://immersive-web.github.io/webxr/#primary-action , each event comes with a frame


There's a concept of a non-animation frame "frame", which is basically for updating input poses.